### ISSUE TITLE
Is it possible to use JSX for templating?

### ISSUE THREAD

**Issue by Azein** [catberry/catberry#313]
After transition to JSX with React, I'm kinda reluctant to use templating engines. But I like Catberry's way to do isomorphic stuff - are there any solutions with JSX? Maybe a setup guide/demo, because it won't be so trivial, probably.

**Comment by pragmader** [comment]
TLDR;
No, we can't use JSX in Catberry.

The way Catberry works, it can theoretically use any template engine if you can implement the [TemplateProvider](https://github.com/catberry/catberry-handlebars/blob/develop/browser/TemplateProvider.js) for it (for server, you need an additional method to [compile](https://github.com/catberry/catberry-handlebars/blob/develop/lib/TemplateProvider.js) the templates).

So, the basic idea is to compile any template to a string during the bundle build process. After it becomes a part of the browser bundle you can use the compiled string to make the template active in a browser.

JSX was created to mix JavaScript logic and DOM representation, we're not able to just compile it to a string and then activate in a browser because it requires all the source code around it. I personally still don't like the whole idea behind JSX.

One of the Catberry's goals is to have a dedicated place for a template and don't mix it with the source code. It keeps projects more organized and less confusing IMO.

**Comment by Azein** [comment]
Okay, thanks for the answer.
